packages/ubuntu_desktop_installer/lib/utils/product_info_extractor.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:io';
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:flutter/foundation.dart';
+
+class ProductInfoExtractor {
+  static String _cachedProductInfo = '';
+
+  final FileSystem fileSystem;
+
+  ProductInfoExtractor({
+    this.fileSystem = const LocalFileSystem(),
+  });
+
+  String getProductInfo({
+    String isoPath = '/cdrom/.disk/info',
+    String localPath = '/etc/os-release',
+  }) {
+    if (_cachedProductInfo.isNotEmpty) {
+      return _cachedProductInfo;
+    }
+
+    if (fileSystem.file(isoPath).existsSync()) {
+      try {
+        final content = fileSystem
+            .file(isoPath)
+            .readAsLinesSync()
+            .firstWhere((line) => line.trim().isNotEmpty);
+
+        _cachedProductInfo = content.substring(0, content.indexOf('"') - 1);
+      } on Exception catch (e) {
+        _extractLocalVersion(localPath);
+      }
+    } else {
+      _extractLocalVersion(localPath);
+    }
+
+    return _cachedProductInfo;
+  }
+
+  void _extractLocalVersion(String localPath) {
+    if (fileSystem.file(localPath).existsSync()) {
+      try {
+        final content = fileSystem
+            .file(localPath)
+            .readAsLinesSync()
+            .firstWhere((line) => line.startsWith('PRETTY_NAME'));
+
+        _cachedProductInfo = content.substring(
+            content.indexOf("\"") + 1, content.lastIndexOf("\""));
+      } on Exception catch (e) {
+        _cachedProductInfo = 'Ubuntu';
+      }
+    } else {
+      _cachedProductInfo = 'Ubuntu';
+    }
+  }
+
+  @visibleForTesting
```

Comment:
I added this because there's no other way to reset the state and I don't want to flaky tests to happen

Reply:
Unfortunately this annotation only provides an info-level message from the analyzer, it doesn't forbid calling the method in production code, so it feels hackish. Maybe add a `force_reload` (or `ignore_cached`) parameter to `getProductInfo()`?